SourceX/storm/storm_dx.cpp

```diff
@@ -27,6 +29,13 @@ BOOL SDrawUpdatePalette(unsigned int firstentry, unsigned int numentries, PALETT
 		return false;
 	}
 
+#ifdef USE_SDL1
+	if (SDL_SetPalette(pal_surface, SDL_LOGPAL, palette->colors, 0, palette->ncolors) != 1) {
```

Comment:
Couldn't this be called from within `SDL_SetPaletteColors`?

Reply:
For that to work, the `SDL_SetPaletteColors` would have to know about the `pal_surface`. I don'd think it should know about that.

I've added a detailed comment explaining what's going on here.

In SDL2 the Surface's palette is independent of the Surface, e.g. you can assign the same palette to multiple surfaces, modify it, and all the surfaces will then immediately use the "new" palette.

In SDL1 the palette is owned by the Surface, and the `SDL_SetPalette`

This is what I've gathered from reading the code of both SDL versions. It's my first time working with SDL though.

Reply:
Ok, currently use the palette in the same way as you describe for SDL1. I don't think we will be taking advantage of the SDL2 feature and instead simply decode the sprites to the systems native format on load, so I'll leave it up to you for what you want to do here.

Reply:
Debugging it would be hell though if someone ever calls `SDL_SetPaletteColors` for a palette not used by `pal_surface`, so I'd prefer with stick with this `ifdef`.